protocol/message_persistence.go

```diff
@@ -1891,6 +1891,58 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
 	return countWithMentions + countNoMentions, countWithMentions, err
 }
 
+func (db sqlitePersistence) MarkMessageAsUnread(chatID string, messageId string) (uint64, uint64, error) {
+	tx, err := db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return 0, 0, err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	_, err = tx.Exec(`UPDATE user_messages SET seen = 0 WHERE seen = 1 AND (mentioned OR replied) AND id = ?`, messageId)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	var countWithMentions uint64
+	row := tx.QueryRow("SELECT changes();")
+	if err := row.Scan(&countWithMentions); err != nil {
+		return 0, 0, err
+	}
+
+	_, err = tx.Exec(`UPDATE user_messages SET seen = 0 WHERE seen = 1 AND NOT(mentioned) AND NOT(replied) AND id = ?`, messageId)
```

Comment:
I think the reason why the count of unread mentions doesn't update correctly is because you are only marking as unread the one message you selected.

I think you're only missing a SELET to get the timestamp of the message that was clicked, then instead of updating using the `id`, use a `> timestamp` to make sure all newer messages as **also** marked as unread!

Reply:
> I think the reason why the count of unread mentions doesn't update correctly is because you are only marking as unread the one message you selected.
> 
> I think you're only missing a SELET to get the timestamp of the message that was clicked, then instead of updating using the `id`, use a `> timestamp` to make sure all newer messages as **also** marked as unread!

oh... that's interesting perspective. Thanks, let me have a look